src/components/buttons/_private/ButtonBase/ButtonBase.tsx

```diff
@@ -54,6 +59,10 @@ export interface IButtonBaseProps extends IButtonCommonProps {
 	padding?: ButtonPropPadding | keyof typeof ButtonPropPadding;
 	/** The font-fontSize applied to the button. */
 	fontSize?: ButtonPropFontSize | keyof typeof ButtonPropFontSize;
+	/** The font-weight applied to the button */
+	fontWeight?: ButtonPropFontWeight;
+	/** The casing applied to the button */
+	fontCasingTransform: boolean;
```

Comment:
Assuming this is needed for something, I'd like to either rethink its naming or change it from a boolean to a list of text-transform css options.

Reply:
I'm going to go for a list of values that can then be tapped into with `privateOptions`. That way, we can get flexibility on the Local side as well as make this easily extensible in the future.

Reply:
This way also better matches the pattern that I used for `fontWeight`